mixer/pkg/api/grpcServer.go

```diff
@@ -157,17 +158,24 @@ func (s *grpcServer) check(ctx context.Context, req *mixerpb.CheckRequest,
 			ValidUseCount:        cr.ValidUseCount,
 			Status:               cr.Status,
 			ReferencedAttributes: protoBag.GetReferencedAttributes(s.globalDict, globalWordCount),
+			RouteDirective:       cr.RouteDirective,
 		},
 	}
 
 	if s.cache != nil {
 		// keep this for later...
+		var directive mixerpb.RouteDirective
+		if cr.RouteDirective != nil {
+			directive = *cr.RouteDirective
+		}
+
 		s.cache.Set(protoBag, checkcache.Value{
 			StatusCode:           resp.Precondition.Status.Code,
 			StatusMessage:        resp.Precondition.Status.Message,
 			Expiration:           time.Now().Add(resp.Precondition.ValidDuration),
 			ValidUseCount:        resp.Precondition.ValidUseCount,
 			ReferencedAttributes: *resp.Precondition.ReferencedAttributes,
+			RouteDirective:       directive,
```

Comment:
My intent was to make GC's work easier here. If the struct is packed into the cache entry, then it can be reclaimed as a unit. I really don't know how big of a deal it is, but it looked like other fields in the struct had similar logic applied.